### PR TITLE
[Feature] Support Pathlib for Config.fromfile

### LINKED ISSUE #7649: make `mmdet.apis.init_detector` work with pathlib
**Describe the feature**
make `mmdet.apis.init_detector` work with pathlib 
**Motivation**
Since mmcv works with pathlib, there is no reason to use str path only. (ref https://github.com/open-mmlab/mmcv/issues/3).
https://github.com/open-mmlab/mmdetection/blob/3e2693151add9b5d6db99b944da020cba837266b/mmdet/apis/inference.py#L31

### PR DESCRIPTION
**Pull request by nijkah**
## Motivation

Closes #7649 

## Modification

Fix to allow Path

### CODE REVIEW

File: mmdet/apis/inference.py [hunk 1]
```diff
@@ -18,8 +19,8 @@ def init_detector(config, checkpoint=None, device='cuda:0', cfg_options=None):
     """Initialize a detector from config file.
 
     Args:
-        config (str or :obj:`mmcv.Config`): Config file path or the config
-            object.
+        config (str or :obj:`Path` or :obj:`mmcv.Config`): Config file path or
```
Comment (ZwwWayne) [inline mmdet/apis/inference.py:22]: Thr docstring can be: Config file path, :obj:`Path`, or the config object.

**Review by ZwwWayne — COMMENTED** [review]
Overall, this pr LGTM. Can be merged after fixing the small issue of docstring.

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
> @ZwwWayne Should I add a test code for this?

It would be great if you can add unit tests for this!

**Comment by nijkah** [comment]
@ZwwWayne Currently, it does not seem to test `init_detector` solely.

Where can I put a test code?
https://github.com/open-mmlab/mmdetection/blob/bde7b4b7eea9dd6ee91a486c6996b2d68662366d/tests/test_runtime/test_async.py#L42
https://github.com/open-mmlab/mmdetection/blob/bde7b4b7eea9dd6ee91a486c6996b2d68662366d/tests/test_runtime/async_benchmark.py#L52

I am thinking a very simple test code like
```
str_path = "config_to_path"
model = init_detector(str_path)


path_object = Path("config_to_path")
model = init_detector(path_object)
```

File: tests/test_runtime/test_apis.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    project_dir = os.path.join(project_dir, '..')
+
+    config_file = os.path.join(
+        project_dir, 'configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.py')
+
+    checkpoint_file = os.path.join(
+        project_dir,
+        'checkpoints/mask_rcnn_r50_fpn_1x_coco_20200205-d4b0c5d6.pth')
+
+    if not os.path.exists(checkpoint_file):
```
Comment (ZwwWayne) [inline tests/test_runtime/test_apis.py:22]: Downloading the checkpoint during each CI will cost extra fees. We can skip loading checkpoint in the unit tests.

Comment (nijkah) [reply tests/test_runtime/test_apis.py:22]: I borrowed from here(https://github.com/open-mmlab/mmdetection/blob/bde7b4b7eea9dd6ee91a486c6996b2d68662366d/tests/test_runtime/async_benchmark.py#L39).
But, it looks redundant to me in this unit test. I will remove it

File: tests/test_runtime/test_apis.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    model = init_detector(config_file, device='cpu')
+
+    # test init_detector with config_file: str and cfg_options
+    cfg_options = dict(
+        model=dict(
+            backbone=dict(
+                depth=101,
+                init_cfg=dict(
+                    type='Pretrained', checkpoint='torchvision://resnet101'))))
+    model = init_detector(config_file, device='cpu', cfg_options=cfg_options)
```
Comment (jbwang1997) [inline tests/test_runtime/test_apis.py:26]: I think the below test also tests the case of config_file being str. So we can keep the below one and delete the above one to save unittest time.

We can also change the ResNet50 to ResNet18 in cfg_options.

**Review by jbwang1997 — APPROVED** [review]
LGTM

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**